components/post_view/post_flag_icon/post_flag_icon.js

```diff
@@ -84,45 +82,47 @@ export default class PostFlagIcon extends React.PureComponent {
     }
 
     render() {
-        if (this.props.isEphemeral) {
-            return null;
-        }
-
         const isFlagged = this.props.isFlagged;
 
         const flagVisible = isFlagged ? 'visible' : '';
 
+        const iconClass = 'icon icon--small';
+
         let flagIcon;
         if (isFlagged) {
-            flagIcon = <FlagIconFilled className='icon'/>;
+            flagIcon = <FlagIconFilled className={iconClass + ' icon--small-filled'}/>;
         } else {
-            flagIcon = <FlagIcon className='icon'/>;
+            flagIcon = <FlagIcon className={iconClass}/>;
         }
 
         return (
-            <button
-                ref={this.buttonRef}
-                id={`${this.props.location}_flagIcon_${this.props.postId}`}
-                aria-label={isFlagged ? localizeMessage('flag_post.unflag', 'Unflag').toLowerCase() : localizeMessage('flag_post.flag', 'Flag for follow up').toLowerCase()}
-                className={'style--none flag-icon__container ' + flagVisible}
-                onClick={this.handlePress}
+            <OverlayTrigger
+                className='hidden-xs'
+                key={'flagtooltipkey' + flagVisible}
+                delayShow={Constants.OVERLAY_TIME_DELAY}
+                placement='top'
+                overlay={
+                    <Tooltip
+                        id='flagTooltip'
+                        className='hidden-xs'
+                    >
+                        <FormattedMessage
+                            id={isFlagged ? t('flag_post.unflag') : t('flag_post.flag')}
+                            defaultMessage={isFlagged ? 'Remove from Saved' : 'Save'}
+                        />
+                    </Tooltip>
+                }
             >
-                <OverlayTrigger
-                    key={'flagtooltipkey' + flagVisible}
-                    delayShow={Constants.OVERLAY_TIME_DELAY}
-                    placement='top'
-                    overlay={
-                        <Tooltip id='flagTooltip'>
-                            <FormattedMessage
-                                id={isFlagged ? t('flag_post.unflag') : t('flag_post.flag')}
-                                defaultMessage={isFlagged ? 'Unflag' : 'Flag for follow up'}
-                            />
-                        </Tooltip>
-                    }
+                <button
+                    ref={this.buttonRef}
+                    id={`${this.props.location}_flagIcon_${this.props.postId}`}
+                    aria-label={isFlagged ? localizeMessage('flag_post.unflag', 'Remove from Saved').toLowerCase() : localizeMessage('flag_post.flag', 'Save').toLowerCase()}
+                    className={'post-menu__item ' + flagVisible}
```

Comment:
3/5 prefer the use of 'classnames'

Reply:
Figured out that the `visible` class is no longer needed, so refactored the code and removed it. Hence, no need for multiple classes or classNames anymore 🙂